### PR TITLE
Configuration [WIP]

### LINKED ISSUE #178: Config via DSN Query String

### PR DESCRIPTION
**Pull request by mitchellhenke**
Centralized configuration, and will be adding something like https://github.com/getsentry/sentry-java/blob/master/sentry/src/main/java/io/sentry/config/Lookup.java#L86-L152 where we'll check env, then DSN, else use default value.

Will eventually close #178 and #179

### CODE REVIEW

File: lib/sentry/config.ex [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  @default_hackney_timeout 5000
+  @default_exclude_patterns [~r"/_build/", ~r"/deps/", ~r"/priv/"]
+  @default_path_pattern "**/*.ex"
+  @default_context_lines 3
+  @default_sample_rate 1.0
+
+  def validate_config! do
+  end
+
+  def get_included_environments do
```
Comment (jeregrine) [inline lib/sentry/config.ex]: Wondering if we need the `get_` prefix.

File: lib/sentry/config.ex [hunk 2]
```diff
@@ -0,0 +1,89 @@
+defmodule Sentry.Config do
```
Comment (jeregrine) [inline lib/sentry/config.ex]: I'm wondering if we load the config once at boot|compile and then return a struct with all the values set? Or do we need to hit the Application env ets table every time?

File: lib/sentry/config.ex (hunk 1 above)
Comment (mitchellhenke) [inline lib/sentry/config.ex]: yeah, wasn't sure about that.  will drop

Comment (mitchellhenke) [inline lib/sentry/config.ex]: I can't really see the module doing any updating, etc., so it should be safe to assume we're only ever getting.

File: lib/sentry/config.ex (hunk 2 above)
Comment (mitchellhenke) [inline lib/sentry/config.ex]: Have thought about it, but would prevent people updating config at runtime.  I'm not sure on the popularity of doing that.

File: lib/mix/tasks/sentry.send_test_event.ex [hunk 3]
```diff
@@ -21,8 +22,8 @@ defmodule Mix.Tasks.Sentry.SendTestEvent do
     Mix.shell.info "public_key: #{public_key}"
     Mix.shell.info "secret_key: #{secret_key}"
     Mix.shell.info "included_environments: #{inspect included_environments()}"
-    Mix.shell.info "current environment_name: #{inspect environment_name()}"
-    Mix.shell.info "hackney_opts: #{inspect hackney_opts()}\n"
+    Mix.shell.info "current environment_name: #{inspect Config.environment_name()}"
```
Comment (jeregrine) [inline lib/mix/tasks/sentry.send_test_event.ex]: Should this function be moved into Config? Config.print_info()

File: lib/sentry/config.ex (hunk 2 above)
Comment (samphilipd) [inline lib/sentry/config.ex]: Please continue to allow us to set configuration at runtime :)

Comment (mitchellhenke) [inline lib/sentry/config.ex]: @samphilipd thanks for the feedback.  this should still be the case.  was there a regression that broke runtime configuration changes?

Comment (samphilipd) [inline lib/sentry/config.ex]: @mitchellhenke no, its working fine. I meant, please don't change the current, working behaviour :)

Comment (mitchellhenke) [inline lib/sentry/config.ex]: @samphilipd ah got it! thanks again 🙂

**Pull request merged**